src/server/system_services/account_server.js

```diff
@@ -803,15 +803,34 @@ async function update_external_connection(req) {
         throw new RpcError('INVALID_CREDENTIALS', `Credentials are not valid ${name}`);
     }
 
+    let set_obj;
+    if (azure_log_access_keys?.azure_tenant_id?.unwrap()) {
```

Comment:
having Azure log access keys is a very small scenario and I think that the code flow should be:
1. create set_obj // -> main flow first
2. if (azure_log_access_keys) set the azure log fields on set_obj // -> other flows